fendermint/vm/interpreter/src/fvm/upgrade_scheduler.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::collections::BTreeMap;
+
+use fvm_ipld_blockstore::Blockstore;
+
+use super::state::{snapshot::BlockHeight, FvmExecState};
+
+/// a function type for migration
+// TODO: Add missing parameters
+pub type MigrationFunc<DB> = fn(state: &mut FvmExecState<DB>) -> anyhow::Result<()>;
+
+pub type ChainID = u64;
```

Comment:
There is a [type for it](https://github.com/consensus-shipyard/ipc/blob/1a72b58a94b8d7efb78a7560e3e5d5a6101269ae/fendermint/vm/core/src/chainid.rs#L7)